packages/amplication-git-service/src/providers/githubProvider.ts

```diff
@@ -0,0 +1,323 @@
+
+  public async commit(
+    branch: string,
+    message: string,
+    files: { path: string; content: string }[],
+    headCommit: string
+  ): Promise<string> {
+    const fileTreeSha = await GithubProvider.createFilesTree(
+      this.octokit,
+      this.owner,
+      this.repo,
+      headCommit,
+      files
+    );
+    const commitSha = await GithubProvider.createCommit(
+      this.octokit,
+      this.owner,
+      this.repo,
+      message,
+      headCommit,
+      fileTreeSha
+    );
+    await GithubProvider.updateRef(
+      this.octokit,
+      this.owner,
+      this.repo,
+      branch,
+      commitSha
+    );
+    return commitSha;
+  }
+
+  public async createPullRequest(
+    title: string,
+    body: string,
+    branch: string,
+    base: string
+  ): Promise<PullRequestMeta> {
+    try {
+      const { html_url: url, number } = (
+        await this.octokit.rest.pulls.create({
+          owner: this.owner,
+          repo: this.repo,
+          title,
+          body,
+          base,
+          head: branch
+        })
+      ).data;
+      return {
+        url,
+        number
+      };
+    } catch (err) {
+      switch (err?.response?.data?.errors[0]?.message) {
+        case `No commits between ${base.toLowerCase()} and ${branch.toLowerCase()}`:
+          throw new Error(`There is no new commit in branch ${branch}`);
+        case `A pull request already exists for ${this.owner}:${branch}.`:
+          throw new Error('Branch already has pull request');
+        default:
+          throw err;
+      }
+    }
+  }
+
+  public async addPullRequestComment(
+    pullRequestNumber: number,
+    body: string
+  ): Promise<string> {
+    try {
+      const { html_url } = (
+        await this.octokit.rest.issues.createComment({
+          owner: this.owner,
+          repo: this.repo,
+          issue_number: pullRequestNumber,
+          body
+        })
+      ).data;
+      return html_url;
+    } catch (err) {
+      switch (err.message) {
+        case 'Not Found':
+          throw new Error(
+            'Cannot add comment to pull request: pull request not found'
+          );
+        default:
+          console.log(err);
+          throw err;
+      }
+    }
+  }
+
+  public async getOpenedPullRequest(branch: string): Promise<PullRequestMeta> {
+    try {
+      const [pullRequest] = (
+        await this.octokit.rest.pulls.list({
+          owner: this.owner,
+          repo: this.repo,
+          head: `${this.owner}:${branch}`,
+          state: 'open'
+        })
+      ).data;
+      if (pullRequest) {
+        return {
+          url: pullRequest.html_url,
+          number: pullRequest.number
+        };
+      } else {
+        return null;
+      }
+    } catch (err) {
+      switch (err.message) {
+        default:
+          throw err;
+      }
+    }
+  }
+
+  public async updatePullRequest(
+    number: number,
+    state: boolean
+  ): Promise<void> {
+    await this.octokit.rest.pulls.update({
```

Comment:
Missing error handling

Reply:
this is a utility function for the tests, currently, I'm ok with this handled in higher levels